src/main/scala/nl/knaw/dans/easy/export/JSON.scala

```diff
@@ -0,0 +1,93 @@
+/** *****************************************************************************
+  * Copyright 2015 DANS - Data Archiving and Networked Services
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  * *****************************************************************************/
+package nl.knaw.dans.easy.export
+
+import java.io.File
+
+import org.json4s.JsonAST
+import org.json4s.JsonAST.JObject
+import org.json4s.JsonDSL._
+import org.json4s.native.JsonMethods._
+
+import scala.util.{Failure, Try}
+import scala.xml.{Elem, Node}
+
+object JSON {
+
+  /** Creates a JSON structure while replacing some fedora ids with SDO directories on the flight.
+    * 'Some' being objects of an EASY-dataset.
+    *
+    * @param sdoDir a directory for the staged digital object
+    * @param datastreams subsection of foXML
+    * @param relsExt the content of the RESL-EXT datastream
+    * @param s configuration required to construct an sdoDir from a fedora-id
+    * @return a string representation of a json object
+    */
+  def apply(sdoDir: File,
+            datastreams: Seq[Node],
+            relsExt: Elem
+           )(implicit s: Settings): Try[String] = Try {
+
+    val descriptionNode = (relsExt \ "Description").head
+    val objectId = descriptionNode.attribute(rdf, "about").get.head.text.replaceAll("[^/]*/", "")
+    val relations = descriptionNode.descendant.filter(_.label != "#PCDATA")
```

Comment:
Aha, I'm comparing apples with oranges. The code under review just selects bits and pieces from an XML document. The cited code is typically needed when traversing all the nodes of a document and you want to do something with some of the nodes, for example a transformer.